src/framework/components/screen/component.js

```diff
@@ -126,7 +126,7 @@ class ScreenComponent extends Component {
         // the combined scale is 1 for an even blend
         const lx = Math.log2(resolution.x / referenceResolution.x);
         const ly = Math.log2(resolution.y / referenceResolution.y);
-        return Math.pow(2, (lx * (1 - this._scaleBlend) + ly * this._scaleBlend));
+        return Math.pow(2, (lx * (1 - this._scaleBlend) + ly * this._scaleBlend)) || 1;
```

Comment:
I would probably prefer to change the two lines above to avoid the NaN completely, could you test that please.
```
const lx = Math.log2(resolution.x / (referenceResolution.x || 1));
```

Reply:
The original issue is caused by `resolution` but I did a few more tests with `referenceResolution` which is user defined and a negative value can also break the 2D screen.

For completeness, here is what happens at moment:
```
                          |    resolution.x < 0    |    resolution.x = 0   |   resolution.x > 0
--------------------------------------------------------------------------------------------------
referenceResolution.x < 0 | OK (but can't happen?) | NaN if scaleBlend = 1 |         NaN
                          |                        |      0 otherwise      |
--------------------------------------------------------------------------------------------------
referenceResolution.x = 0 |           NaN          |          NaN          | NaN if scaleBlend = 1
                          |                        |                       |  Infinity otherwise
--------------------------------------------------------------------------------------------------
referenceResolution.x > 0 |           NaN          | NaN if scaleBlend = 1 |     OK (obviously)
                          |                        |      0 otherwise      |
```
And same for `y` components too except for `scaleBlend` condition (0 instead of 1).

It should be noted that when it returns 0, it also breaks the 2D screen while Infinity works (though it doesn't show anything until it's back to a more usual value).

I'm not sure how negative and 0 values in `referenceResolution` should be handled (I don't see how `resolution` could be negative at all but maybe it should be considered too), but at the moment nothing prevents the user to do that (not even a warning in the editor) so at least it should be possible to recover from it.

Reply:
I would not worry about the case the user sets a referenceResolution to a negative value. You can pass incorrect values to many things in the engine and break it completely. We only need to handle the resolution being 0 really.

Reply:
Ok, resolution || 1 then.